src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java

```diff
@@ -84,6 +87,16 @@ public AdyenConfigProperties(final Properties properties) {
         this.hppTarget = properties.getProperty(PROPERTY_PREFIX + "hpp.target");
         this.hppSkipDetailsTarget = this.hppTarget != null ? this.hppTarget.replace(this.hppTarget.substring(this.hppTarget.lastIndexOf('/') + 1), "skipDetails.shtml") : null;
         this.hppVariantOverride = properties.getProperty(PROPERTY_PREFIX + "hppVariantOverride");
+
+        int hppExpirationPeriodInDays = 0;
+        try {
+            final String hppExpirationPeriod = properties.getProperty(PROPERTY_PREFIX + "hppExpirationPeriodInDays");
+            hppExpirationPeriodInDays = Integer.parseInt(hppExpirationPeriod);
+        } catch(NumberFormatException e) {
+            hppExpirationPeriodInDays = DEFAULT_HPP_EXPIRATION_PERIOD_IN_DAYS;
+        }
+        this.hppExpirationPeriodInDays = hppExpirationPeriodInDays;
```

Comment:
A more complete implementation would even have per payment method policies. For example, an hour or so should be enough for a PayPal payment, but we'd need a few days for a Boleto.

Not sure it's worth going there though. I would maybe have a more aggressive default though (e.g. 3 days).